src/Request.php

```diff
@@ -109,6 +109,9 @@ public function transfer()
         curl_setopt_array($curl, $this->options);
         // excute the curl and write response into $response
         $response = curl_exec($curl);
+        if (!$response) {
```

Comment:
i think the curl exec response should make a type check compare: if ($response === false) {}